modules/virtual_network_gateway/main.tf

```diff
@@ -1,86 +1,119 @@
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip
+resource "azurerm_public_ip" "this" {
+  for_each = { for ip_configuration in var.ip_configuration : ip_configuration.name => ip_configuration.name if ip_configuration.create_public_ip }
+
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  name                = each.value
+
+  allocation_method = "Static"
+  sku               = "Standard"
+  zones             = try(length(var.zones) > 0, false) ? var.zones : null
```

Comment:
cannot this be just `var.zones`? default is null and everything else is governed by the validation block

Reply:
done